waku-bindings/src/node/events.rs

```diff
@@ -0,0 +1,79 @@
+//! Waku message [event](https://rfc.vac.dev/spec/36/#events) related items
+//!
+//! Asynchronous events require a callback to be registered.
+//! An example of an asynchronous event that might be emitted is receiving a message.
+//! When an event is emitted, this callback will be triggered receiving an [`Event`]
+
+// std
+use std::ffi::c_void;
+// crates
+use serde::{Deserialize, Serialize};
+// internal
+use crate::general::WakuMessage;
+use crate::utils::get_trampoline;
+use crate::MessageId;
+
+/// Waku event
+/// For now just WakuMessage is supported
+#[non_exhaustive]
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "eventType", rename_all = "camelCase")]
+pub enum Event {
+    #[serde(rename = "message")]
+    WakuMessage(WakuMessageEvent),
+    Unrecognized(serde_json::Value),
+}
+
+/// Type of `event` field for a `message` event
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct WakuMessageEvent {
+    /// The pubsub topic on which the message was received
+    pubsub_topic: String,
+    /// The message id
+    message_id: MessageId,
+    /// The message in [`WakuMessage`] format
+    waku_message: WakuMessage,
+}
+
+impl WakuMessageEvent {
+    pub fn pubsub_topic(&self) -> &String {
+        &self.pubsub_topic
+    }
+
+    pub fn message_id(&self) -> &String {
+        &self.message_id
+    }
+
+    pub fn waku_message(&self) -> &WakuMessage {
+        &self.waku_message
+    }
+}
```

Comment:
I've not seen that often in Rust code. Maybe use `pub` struct fields?

Reply:
Also, never use `&String`, usually what you want is `&str`

Reply:
Fixed in https://github.com/waku-org/waku-rust-bindings/pull/87/commits/c434e9ebf59afd45f6ebffce7e7d7470e69729e5